hydrogym/ts.py

```diff
@@ -143,18 +157,65 @@ def initialize_operators(self):
             proj_prob, solver_parameters={"ksp_type": "cg", "pc_type": "sor"}
         )
 
+    def set_damping(self, k_new):
+        if not isinstance(k_new, list):
+            k_new = [k_new]
+        self.controller_damping_coeff = k_new
+        return
+
+    def get_inertia(self):
+        return self.inertia
+
+    def get_state(self):
+        return self.state
+
     def update_controls(self, control):
-        """Add a damping factor to the controller response
+        if self.control_method == "indirect":
+            if self.flow.name == "Cylinder":
+                """
+                Update BCS of cylinder to new angular velocity using implicit euler solver according to diff eqn:
+
+                omega_t[i+1] = omega_t[i] + (d_omega/dt)_t[i+1] * dt
+                            = omega_t[i] + (control_t[i+1] - k_damping*omega_t[i+1])/I_cm * dt
+
+                omega_t[i+1] can be solved for directly in order to avoid using a costly root solver
+                """
+                next_state = []
+                for (state, ctrl, I_cm, k_damping) in zip(
+                    self.state, control, self.I_cm, self.k_damping
+                ):
+                    next_state.append(
+                        (state + ctrl * self.dt / I_cm)
+                        / (1 + k_damping * self.dt / I_cm)
+                    )
+                self.state = next_state
```

Comment:
This is nitpicky, but can we avoid the list `next_state` by just assigning `state = ...` within the loop?